#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.49325135116178986 -0.8374957594503853 -0.23516793463329766 5.2746500835210695
  position 0.8588060096015457 0.5713593792847367 0.5576181325304683
  follow "robot:solid:solid:solid:hand"
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
  floorTileSize 0.2 0.2
}
Robot {
  children [
    DEF BASE Group {
      children [
        Transform {
          translation 0 0.01 0
          children [
            Shape {
              appearance DEF solidColor PBRAppearance {
                baseColor 0 0 1
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.02
                radius 0.1
              }
            }
          ]
        }
        Transform {
          translation 0 0.025 0
          children [
            Shape {
              appearance DEF rotationColor PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.03
              }
            }
          ]
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.025 0
      }
      device [
        DEF pos1 PositionSensor {
          name "posMotor1"
        }
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint Solid {
        translation 0 0.025 0
        children [
          DEF arm1 Group {
            children [
              Transform {
                translation 0 0.05 0
                children [
                  Shape {
                    appearance USE solidColor
                    geometry Cylinder {
                      height 0.1
                      radius 0.05
                    }
                  }
                ]
              }
              Transform {
                translation 0 0.08 0
                rotation 1 0 0 1.5708
                children [
                  Shape {
                    appearance USE rotationColor
                    geometry Cylinder {
                      height 0.14
                      radius 0.02
                    }
                  }
                ]
              }
            ]
          }
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
              anchor 0 0.08 0
            }
            device [
              PositionSensor {
                name "posMotor2"
              }
              RotationalMotor {
                name "motor2"
              }
            ]
            endPoint Solid {
              translation 0 0.08 0
              rotation 0 0 -1 1.5707999999999998
              children [
                DEF arm2 Group {
                  children [
                    Transform {
                      translation 0 0.13 0
                      children [
                        Shape {
                          appearance USE solidColor
                          geometry Cylinder {
                            height 0.12
                            radius 0.05
                          }
                        }
                      ]
                    }
                    Transform {
                      translation 0 0.09 0
                      rotation 1 0 0 1.5708
                      children [
                        Shape {
                          appearance USE rotationColor
                          geometry Cylinder {
                            height 0.14
                            radius 0.02
                          }
                        }
                      ]
                    }
                    Transform {
                      translation 0 0.17 0
                      rotation 1 0 0 1.5708
                      children [
                        Shape {
                          appearance USE rotationColor
                          geometry Cylinder {
                            height 0.14
                            radius 0.02
                          }
                        }
                      ]
                    }
                    Transform {
                      translation 0 0.05 -0.058
                      children [
                        Shape {
                          appearance USE rotationColor
                          geometry Box {
                            size 0.034 0.1 0.02
                          }
                        }
                      ]
                    }
                    Transform {
                      translation 0 0.05 0.058
                      children [
                        Shape {
                          appearance USE rotationColor
                          geometry Box {
                            size 0.034 0.1 0.02
                          }
                        }
                      ]
                    }
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0 0 1
                    anchor 0 0.17 0
                  }
                  device [
                    PositionSensor {
                      name "posMotor3"
                    }
                    RotationalMotor {
                      name "motor3"
                    }
                  ]
                  endPoint Solid {
                    translation 0 0.17 0
                    rotation 0 0 1 0
                    children [
                      DEF arm3 Group {
                        children [
                          Transform {
                            translation 0 0.13 0
                            children [
                              Shape {
                                appearance USE solidColor
                                geometry Cylinder {
                                  height 0.12
                                  radius 0.05
                                }
                              }
                            ]
                          }
                          Transform {
                            translation 0 0.09 0
                            rotation 1 0 0 1.5708
                            children [
                              Shape {
                                appearance USE rotationColor
                                geometry Cylinder {
                                  height 0.14
                                  radius 0.02
                                }
                              }
                            ]
                          }
                          Transform {
                            translation 0 0.05 -0.058
                            children [
                              Shape {
                                appearance USE rotationColor
                                geometry Box {
                                  size 0.034 0.1 0.02
                                }
                              }
                            ]
                          }
                          Transform {
                            translation 0 0.05 0.058
                            children [
                              Shape {
                                appearance USE rotationColor
                                geometry Box {
                                  size 0.034 0.1 0.02
                                }
                              }
                            ]
                          }
                        ]
                      }
                      HingeJoint {
                        jointParameters HingeJointParameters {
                        }
                        device [
                          RotationalMotor {
                            name "armRot"
                          }
                        ]
                        endPoint Solid {
                          translation 0 0.2 0
                          rotation -0.5773509358554485 0.5773509358554485 0.5773489358556708 2.0944
                          children [
                            DEF GRAPBASE Group {
                              children [
                                Transform {
                                  children [
                                    Shape {
                                      appearance DEF BLEU PBRAppearance {
                                        baseColor 0 0 1
                                        roughness 1
                                        metalness 0
                                      }
                                      geometry Box {
                                        size 0.03 0.02 0.03
                                      }
                                    }
                                  ]
                                }
                                Transform {
                                  translation 0 0 0.03
                                  children [
                                    Shape {
                                      appearance DEF rouge PBRAppearance {
                                        baseColor 1 0 0
                                        roughness 1
                                        metalness 0
                                      }
                                      geometry Box {
                                        size 0.025 0.015 0.03
                                      }
                                    }
                                  ]
                                }
                              ]
                            }
                            SliderJoint {
                              jointParameters JointParameters {
                                axis 1 0 0
                              }
                              device [
                                LinearMotor {
                                  name "armRight"
                                  maxForce 100
                                }
                                PositionSensor {
                                  name "posArmRight"
                                }
                              ]
                              endPoint Solid {
                                translation -0.023 0 0.032
                                children [
                                  DEF RIGHT Group {
                                    children [
                                      Transform {
                                        rotation 0 0 1 1.5708
                                        children [
                                          Shape {
                                            appearance USE rouge
                                            geometry Cylinder {
                                              height 0.03
                                              radius 0.005
                                            }
                                          }
                                        ]
                                      }
                                      Transform {
                                        translation -0.02 0 0.032
                                        children [
                                          Shape {
                                            appearance USE BLEU
                                            geometry Box {
                                              size 0.01 0.035 0.08
                                            }
                                          }
                                        ]
                                      }
                                    ]
                                  }
                                  TouchSensor {
                                    translation -0.02 0 0.04
                                    children [
                                      DEF touchRight Shape {
                                        appearance Appearance {
                                        }
                                        geometry Box {
                                          size 0.01 0.02 0.04
                                        }
                                      }
                                    ]
                                    name "sensRight"
                                    boundingObject USE touchRight
                                    physics Physics {
                                      density 0.1
                                    }
                                  }
                                ]
                                name "RIGHT"
                                boundingObject USE RIGHT
                                physics Physics {
                                  density -1
                                  mass 0.05
                                }
                              }
                            }
                            SliderJoint {
                              jointParameters JointParameters {
                                axis 1 0 0
                              }
                              device [
                                LinearMotor {
                                  name "armLeft"
                                  maxForce 100
                                }
                                PositionSensor {
                                  name "posArmLeft"
                                }
                              ]
                              endPoint Solid {
                                translation 0.023 0 0.032
                                rotation 0 0 1 -3.137485307179586
                                children [
                                  DEF LEFT Group {
                                    children [
                                      Transform {
                                        rotation 0 0 1 1.5708
                                        children [
                                          Shape {
                                            appearance USE rouge
                                            geometry Cylinder {
                                              height 0.03
                                              radius 0.005
                                            }
                                          }
                                        ]
                                      }
                                      Transform {
                                        translation -0.02 0 0.032
                                        children [
                                          Shape {
                                            appearance USE BLEU
                                            geometry Box {
                                              size 0.01 0.035 0.08
                                            }
                                          }
                                        ]
                                      }
                                    ]
                                  }
                                  TouchSensor {
                                    translation -0.02 0 0.04
                                    children [
                                      DEF touchLeft Shape {
                                        appearance Appearance {
                                        }
                                        geometry Box {
                                          size 0.01 0.02 0.04
                                        }
                                      }
                                    ]
                                    name "sensLeft"
                                    boundingObject USE touchLeft
                                    physics Physics {
                                      density 1
                                    }
                                  }
                                ]
                                name "LEFT"
                                boundingObject USE LEFT
                                physics Physics {
                                  density -1
                                  mass 0.05
                                }
                              }
                            }
                          ]
                          name "hand"
                          boundingObject USE GRAPBASE
                          physics Physics {
                            density 10
                          }
                        }
                      }
                    ]
                    boundingObject USE arm3
                    physics Physics {
                      density 10
                    }
                  }
                }
              ]
              boundingObject USE arm3
              physics Physics {
                density 10
              }
            }
          }
        ]
        boundingObject USE arm1
        physics Physics {
        }
      }
    }
  ]
  boundingObject USE GRAPBASE
  controller "armController"
}
